lib/ButtonNew/ButtonIcon/ButtonContainedIcon.js

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react';
+import { ButtonIcon } from 'lib';
+import { createClassNames } from 'helpers/createClassNames';
+import { buttonIconDefaultProps } from '../common';
+
+function ButtonContainedIcon(props) {
+  const [active, setActive] = useState(props.active);
+  const classNames = createClassNames('button-contained-icon', props, [
+    'border border-solid border-neutral-90 bg-white',
+    'hover:bg-blue-90 hover:border-blue-primary'
+  ]);
+
+  return (
+    <ButtonIcon
+      onMouseOver={() => setActive(true)}
+      onFocus={() => setActive(true)}
+      onMouseOut={() => setActive(false)}
+      onBlur={() => setActive(false)}
```

Comment:
🚩 This feels weird, especially if its just for hover and focus styling. This component feels like instead of using ButtonIcon as a base you might have been better just using ButtonBase and building from the ground up. 

Happy to pair on this if im just missing some obvious issue that this solves!

Reply:
From GCDS icon buttons should have always had a "contained" pattern. 

This was missed out (probably due to speed and getting `ButtonIcon` out the door) and so without this abstraction, it becomes a task of refactoring the base component (maybe even the entire NEW button family) and testing our entire usage of said component. 🤦 

Now, I'm not sure why but we HAVE to pass `active` as a prop to get the icon blue 🤷 So this abstraction is following the patterns set by the original (and quite recent) base component `ButtonIcon`.

----

I'm happy to pair but we can't create a brand new component for this as that won't map to GCDS. 

Maybe for `ButtonIcon`, we should use a component stylesheet cause it's a mess of tailwind restricts I feel atm, for example, [this](https://share.getcloudapp.com/ApuL9bnN) isn't a nice implementation to work with vs using an `scss` file to create a clear direction for states was varients.